packages/optimizely-cms-sdk/src/model/contentTypes.ts

```diff
@@ -35,29 +35,31 @@ type BaseContentType = {
 export type PageContentType = BaseContentType & {
   baseType: '_page';
   mayContainTypes?: Array<
-    ContentType<PageContentType | ExperienceContentType | FolderContentType>
+    | ContentType<PageContentType | ExperienceContentType | FolderContentType>
+    | string
```

Comment:
Would it be possible to validate that the strings are valid content type names? Not sure this is possible in TS, but I believe so. Then we could just accept strings and one would get query completion/IntelliSense.

Reply:
It may be possible but it will require work from the developer. We need to "generate" the type `type ContentTypes = "ct1" | "ct2"...` somehow (maybe infering from the registry) but since the registry is defined by the developer, they need to define the `ContentTypes` type globally somehow.

For example, in PayloadCMS you need to run a command `payload generate: types` to generate it.